firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionServiceImplTest.java

```diff
@@ -106,13 +110,16 @@ public class FirebaseAppDistributionServiceImplTest {
           .setBinaryType(BinaryType.AAB)
           .build();
 
-  private static final AppDistributionReleaseInternal.Builder TEST_RELEASE_NEWER_APK_INTERNAL =
+  private static final AppDistributionReleaseInternal TEST_RELEASE_NEWER_APK_INTERNAL =
       AppDistributionReleaseInternal.builder()
           .setBuildVersion("3")
           .setDisplayVersion("3.0")
           .setReleaseNotes("Newer version.")
           .setBinaryType(BinaryType.APK)
-          .setDownloadUrl(TEST_URL);
+          .setDownloadUrl(TEST_URL)
+          .build();
+
+  private final ExecutorService testExecutor = Executors.newSingleThreadExecutor();
```

Comment:
It's actually preferable to use the actual executor from components as it will detect violations, i.e. disk reads, network io, etc and crash the tests. Otherwise there is no guarantee that the code under test uses the right executor which will slow down all of firebase and cause logcat warnings in developers' apps

Reply:
you can use this utility class to get a hold of the right executor:
https://github.com/firebase/firebase-android-sdk/blob/master/integ-testing/src/main/java/com/google/firebase/concurrent/TestOnlyExecutors.java

Reply:
Oh, that's awesome. Done.